CHANGELOG.md

```diff
@@ -6,10 +6,13 @@ _The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 
 ## Unreleased
 
+- Fix memory leak issue when using component selector styles. (see [#2424](https://github.com/styled-components/styled-components/issues/2424)).
```

Comment:
I noticed the comment saying to update this and I agree this isn't a leak, but helps prevent unnecessary GC cycles.